index.bs

```diff
@@ -376,42 +408,75 @@ To <dfn>parse an attribution destination</dfn> from a string |str|:
 
 <h3 algorithm id="obtaining-attribution-source-anchor">Obtaining an attribution source from an <code>a</code> element</h3>
 
-To <dfn>obtain an attribution source</dfn> from an <{a}> element |anchor|:
+To <dfn>obtain an attribution source from an anchor</dfn> given an <{a}> element |anchor|:
+
+1. Let |params| be an {{AttributionSourceParams}} with the following key/value pairs:
+     : {{AttributionSourceParams/attributionDestination}}
+     :: |anchor|'s <{a/attributiondestination}> attribute
+     : {{AttributionSourceParams/attributionSourceEventId}}
+     :: |anchor|'s <{a/attributionsourceeventid}> attribute
+     : {{AttributionSourceParams/attributionReportTo}}
+     :: |anchor|'s <{a/attributionreportto}> attribute
+     : {{AttributionSourceParams/attributionExpiry}}
+     :: |anchor|'s <{a/attributionexpiry}> attribute
+     : {{AttributionSourceParams/attributionSourcePriority}}
+     :: |anchor|'s <{a/attributionsourcepriority}> attribute
+1. Return the result of running [=obtain an attribution source from params=] with |params|.
+
+<h3 algorithm id="obtaining-event-attribution-source-anchor">Obtaining an event attribution source from an <code>a</code> element</h3>
+
+To <dfn>maybe obtain an event attribution source from an anchor</dfn> given an <{a}> element |anchor|, run the following steps:
+
+1. If |anchor| does not have a <{a/registerattributionsource}> attribute, return null.
+1. Let |source| be the result of running [=obtain an attribution source from an anchor=] with |anchor|.
+1. If |source| is null, return null.
+1. Return the result of running [=obtain an event attribution source from a source=] with |source|.
+
+<h3 algorithm id="obtaining-event-attribution-source-source">Obtaining an event attribution source from a source</h3>
+
+To <dfn>obtain an event attribution source from a source</dfn> given an [=attribution source=] |source|, run the following steps:
+
+1. Set |source|'s [=attribution source/source type=] to "<code>event</code>".
+1. Round |source|'s [=attribution source/expiry=] away from zero to the nearest day.
+1. Return |source|.
+
+<h3 algorithm id="obtaining-attribution-source-params">Obtaining an attribution source from {{AttributionSourceParams}}</h3>
+
+To <dfn>obtain an attribution source from params</dfn> given an {{AttributionSourceParams}} |params|, run the following steps:
+
 1. Let |sourceIdentifier| be a new unique opaque string.
 1. Let |currentTime| be the current time.
-1. If |anchor| does not have both an <{a/attributiondestination}> attribute and
-    an <{a/attributionsourceeventid}> attribute, return null.
-1. If |anchor|'s [=relevant settings object=]'s [=environment settings object/responsible document=] is
+1. If |params| does not have both an {{AttributionSourceParams/attributionDestination}} field and
+    an {{AttributionSourceParams/attributionSourceEventId}} field, return null.
```

Comment:
These entries are `required`, so you'll have to deal with the possibility that they're missing from the anchor earlier in the flow.